File: snippets/snippets/node/transcriptions-post.js
import { CohereClientV2 } from 'cohere-ai';
import fs from 'fs';

const cohere = new CohereClientV2({});

(async () => {
  const fileBuffer = fs.readFileSync('./sample.wav');
  const file = new File([fileBuffer], 'sample.wav');

  const response = await cohere.audio.transcriptions.create(file, {
    model: 'cohere-transcribe-03-2026',
    language: 'en',
  });

  console.log(response.text);
})();

File: snippets/snippets/java/app/src/main/java/TranscriptionsPost.java
/* (C)2024 */
import com.cohere.api.Cohere;
import com.cohere.api.resources.audio.transcriptions.requests.TranscriptionsCreateRequest;
import com.cohere.api.resources.audio.transcriptions.types.AudioTranscriptionsCreateResponse;
import java.io.File;

public class TranscriptionsPost {
  public static void main(String[] args) {
    Cohere cohere = Cohere.builder().clientName("snippet").build();

    AudioTranscriptionsCreateResponse response =
        cohere
            .audio()
            .transcriptions()
            .create(
                new File("./sample.wav"),
                TranscriptionsCreateRequest.builder()
                    .model("cohere-transcribe-03-2026")
                    .language("en")
                    .build());

    System.out.println(response);
  }
}
